weather_mv/loader_pipeline/metrics.py

```diff
@@ -0,0 +1,80 @@
+import time
+import copy
+import inspect
+from functools import wraps
+import apache_beam as beam
+from apache_beam import DoFn, Pipeline, PCollection
+from apache_beam.options.pipeline_options import PipelineOptions
+from apache_beam.metrics import metric
+
+
+def timeit(func_name: str, keyed_fn: bool = False):
+    """Decorator to add time it takes for an element to be processed by a stage.
+
+    Args:
+        func_name: A unique name of the stage.
+        keyed_fn (optional): This has to be passed true if the input is adding keys to the element.
```

Comment:
Can you add more info about this flag please? Like when it would be used.

Reply:
done